### PR TITLE
DOCS-#7144: Add information about logging from user defined function

### LINKED ISSUE #7144: Add information about logging from user defined or remote functions on modin docs
Have to update https://modin.readthedocs.io/en/stable/usage_guide/advanced_usage/modin_logging.html with information on recommendation to log from user defined  functions.

_Originally posted by @YarShev in https://github.com/modin-project/modin/issues/3122#issuecomment-2034378305_

### PR DESCRIPTION
**Pull request by arunjose696**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #7144 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/usage_guide/advanced_usage/modin_logging.rst [hunk 1]
```diff
@@ -53,3 +53,28 @@ Disable Modin logging like so:
   LogMode.disable()
 
   # User code goes here
+
+Debugging from user defined functions:
+
+
+.. warning:: 
+  If developer attempts to use modin logging in the remote function/UDF for logging lower-level operators as below, multiple log directories(.modin/logs/job_**) would be created for each worker executing the remote function.
```
Comment (YarShev) [inline docs/usage_guide/advanced_usage/modin_logging.rst:61]:
```suggestion
  When attempting to use Modin logging in a remote function/UDF for logging lower-level operators
  as below, multiple log directories ``.modin/logs/job_**`` would be created for each worker executing the remote function/UDF.
```

File: docs/usage_guide/advanced_usage/modin_logging.rst [hunk 2]
```diff
@@ -53,3 +53,28 @@ Disable Modin logging like so:
...
+.. code-block:: python
+
+  def Udf(df, partition_shape, *dfs):
+      from modin.config import LogMode, LogMemoryInterval, LogFileSize
+      LogMode.enable()
+      # User code goes here
+      return
```
Comment (YarShev) [inline docs/usage_guide/advanced_usage/modin_logging.rst:68]: Let's declare a udf function and pass it in map or apply method like the user interacts with pandas API. The udf will be executed in a remote function.

```python
def udf(x):
    logger stuff
    return x + 1

modin_df.map(udf)
```

File: docs/usage_guide/advanced_usage/modin_logging.rst [hunk 3]
```diff
@@ -53,3 +53,28 @@ Disable Modin logging like so:
...
+
+So recommended approach would be to use a different logger as in below snipet, to log from user defined functions that execute on ray workers.
+Below is an an example with ray to log from UDF by just declaring the logger config inside the remote function, Developers can log from UDFs and ray remote functions called in lower level functions as in the below example.
+
+.. code-block:: python
+
+  @ray.remote
```
[the remark this answers is not in the document]
Comment (arunjose696) [reply docs/usage_guide/advanced_usage/modin_logging.rst]: Done.

File: docs/usage_guide/advanced_usage/modin_logging.rst (hunk 2 above)
Comment (arunjose696) [reply docs/usage_guide/advanced_usage/modin_logging.rst]: Done.

File: docs/usage_guide/advanced_usage/modin_logging.rst [hunk 4]
```diff
@@ -53,3 +53,39 @@ Disable Modin logging like so:
   LogMode.disable()
 
   # User code goes here
+
+Debugging from user defined functions:
+
```
Comment (YarShev) [inline docs/usage_guide/advanced_usage/modin_logging.rst:58]:
```suggestion
```

File: docs/usage_guide/advanced_usage/modin_logging.rst [hunk 5]
```diff
@@ -53,3 +53,39 @@ Disable Modin logging like so:
...
+
+.. warning:: 
+  When attempting to use Modin logging in user defined functions that execute in workers for logging lower-level operators
```
Comment (YarShev) [inline docs/usage_guide/advanced_usage/modin_logging.rst:61]: Check indents here. There should be 4 spaces like we have in modin/docs/flow/modin/core/dataframe/algebra.rst:L62.

File: docs/usage_guide/advanced_usage/modin_logging.rst [hunk 6]
```diff
@@ -53,3 +53,39 @@ Disable Modin logging like so:
...
+
+  import modin.pandas as pd
+  def udf(x):
+      from modin.config import LogMode, LogMemoryInterval, LogFileSize
+      LogMode.enable()
+      # User code with modin operations
+      return x+1
+  
+  modin_df = pd.DataFrame([0,1,2,3])
+  modin_df.map(udf)
```
Comment (YarShev) [inline docs/usage_guide/advanced_usage/modin_logging.rst:74]:
```suggestion
  import modin.pandas as pd

  def udf(x):
      from modin.config import LogMode
      
      LogMode.enable()
      
      return x + 1
  
  modin_df = pd.DataFrame([0, 1, 2, 3])
  print(modin_df.map(udf))
```

File: docs/usage_guide/advanced_usage/modin_logging.rst [hunk 7]
```diff
@@ -53,3 +53,39 @@ Disable Modin logging like so:
...
+
+So the **recommended** approach would be to use a different logger as in the below snipet
+to log from user defined functions that execute on Ray workers.
```
Comment (YarShev) [inline docs/usage_guide/advanced_usage/modin_logging.rst:77]:
```suggestion
to log from user defined functions that execute on workers.
```

File: docs/usage_guide/advanced_usage/modin_logging.rst [hunk 8]
```diff
@@ -53,3 +53,39 @@ Disable Modin logging like so:
...
+  import modin.pandas as pd
+  import logging
+  def udf(x):
+      logging.basicConfig(filename='modin_udf.log', level=logging.INFO)
+      logging.info("This log message will be written to modin_udf.log ")
+      # User code goes here
+      return x+1
+
+  modin_df = pd.DataFrame([0,1,2,3])
+  print(modin_df.map(udf))
```
Comment (YarShev) [inline docs/usage_guide/advanced_usage/modin_logging.rst:91]:
```suggestion
  import logging
  import modin.pandas as pd
  
  def udf(x):
      logging.basicConfig(filename='modin_udf.log', level=logging.INFO)
      logging.info("This log message will be written to modin_udf.log ")

      # User code goes here
      return x + 1

  modin_df = pd.DataFrame([0, 1, 2, 3])
  print(modin_df.map(udf))
```

**Review by YarShev — APPROVED** [review]

**Pull request merged**